### PR TITLE
Change links to https

### PR DESCRIPTION
**Pull request by jakelishman**
Now that qutip.org supports https, we change all the links to hosted content, and internal/external links that support it over to https to avoid mixed content.  This does not change any links in the documentation itself; any changes would be overwritten when Sphinx re-ran, and the correct place to fix them is in the generating files anyway.  That said, modern builds of the documentation use https where it's appropriate anyway, there's just a few places in the guides and in docstrings where we have unintentionally hard-coded http links.

### CODE REVIEW

File: _includes/footer.html [hunk 1]
```diff
@@ -1,13 +1,13 @@
 <div class="footer">
-<p>&copy; 2013 and later. <a href="http://nqdl.korea.ac.kr">P.D. Nation</a> & <a href="http://jrjohansson.github.io/research.html">J.R. Johansson</a>. This page was created with <a href="http://twitter.github.io/bootstrap/">bootstrap</a>.</p>
+<p>&copy; 2013 and later. <a href="https://nqdl.korea.ac.kr">P.D. Nation</a> &amp; <a href="https://jrjohansson.github.io/research.html">J.R. Johansson</a>.  This page was created with <a href="https://twitter.github.io/bootstrap/">bootstrap</a>.</p>
```
Comment (nathanshammah) [inline _includes/footer.html:2]:
```suggestion
<p>&copy; 2013 and later. <a href="https://researcher.watson.ibm.com/researcher/view.php?person=ibm-Paul.Nation">P.D. Nation</a> &amp; <a href="https://jrjohansson.github.io/research.html">J.R. Johansson</a>.  This page was created with <a href="https://twitter.github.io/bootstrap/">bootstrap</a>.</p>
```

File: news.html [hunk 2]
```diff
@@ -222,7 +222,7 @@ <h4>QuTiP is Presented in an Invited Talk at the RIKEN Berkeley Workshop on Quan
         <h4>QuTiP is Featured on Nature's Physics Blog</h4>
     </div>
     <div class="col-md-4" style='margin-top: 25px'>
-        <a href="http://blogs.nature.com/onyourwavelength/2019/01/09/the-rise-of-open-source-in-quantum-physics-research/"><p>The Rise of Open-Source in Quantum Physics Research</p></a>
+        <a href="https://blogs.nature.com/onyourwavelength/2019/01/09/the-rise-of-open-source-in-quantum-physics-research/"><p>The Rise of Open-Source in Quantum Physics Research</p></a>
```
Comment (nathanshammah) [inline news.html:225]:
```suggestion
        <a href="http://blogs.nature.com/onyourwavelength/2019/01/09/the-rise-of-open-source-in-quantum-physics-research/"><p>The Rise of Open-Source in Quantum Physics Research</p></a>
```

File: news.html [hunk 3]
```diff
@@ -210,7 +210,7 @@ <h4>QuTiP is Presented at FOSDEM 2019 in Bruxelles, Belgium</h4>
         <h4>QuTiP is Presented in an Invited Talk at the RIKEN Berkeley Workshop on Quantum Information Science</h4>
     </div>
     <div class="col-md-4" style='margin-top: 25px'>
-        <a href="http://rb19.lbl.gov"><p>rb19.lbl.gov</p></a>
+        <a href="https://rb19.lbl.gov"><p>rb19.lbl.gov</p></a>
```
Comment (nathanshammah) [inline news.html:213]:
```suggestion
        <a href="https://conferences.lbl.gov/event/195/timetable/?view=nicecompact"><p>rb19 at lbl.gov</p></a>
```

**Review by nathanshammah — CHANGES_REQUESTED** [review]
- Fixed some links that got outdated, such as the group at U. of Korea for @nonhermitian and other links that are not maintained.
- Suggested to keep http for the websites that do not have the safe protocol, to avoid some browsers getting on a disclaimers page (as the https is not such).

File: news.html (hunk 2 above)
Comment (jakelishman) [reply news.html:225]: Oh, _mostly_ I checked the links as I changed them, but I didn't bother actually clicking through to Nature...  I assumed that they'd be fine.  Looking above, looks like I missed one or two as well.

Not shown in the diff are the other few http links I found that didn't seem safe to change.

**Comment by jakelishman** [comment]
@nathanshammah I commited the changes you suggested.  I'd like to merge this in fairly quickly, because right now we hotlink d3.js and some Google fonts over http, which Firefox and Chrome will both block (hence why the user distribution picture isn't loading right now.)

**Review by nathanshammah — APPROVED** [review]
LGTM! Thanks @jakelishman.

**Comment by jakelishman** [comment]
Confirming this has fixed the mixed content delivery of the main site.  Every asset we serve on qutip.org is now https-enforced.

**Pull request merged**